src/main/java/frc/robot/commands/drive/DriveWithJoysticks.java

```diff
@@ -83,29 +85,28 @@ public void execute() {
         new Pose2d(new Translation2d(), linearDirection)
             .transformBy(GeomUtils.transformFromTranslation(linearMagnitude, 0))
             .getTranslation();
-
-    // Convert to meters per second
-    ChassisSpeeds speeds =
-        new ChassisSpeeds(
-            linearVelocity.getX() * DriveConstants.MAX_LINEAR_SPEED,
-            linearVelocity.getY() * DriveConstants.MAX_LINEAR_SPEED,
-            rightX * MAX_ANGULAR_SPEED);
-
-    // Convert from field relative
-    if (robotRelativeOverride.get()) {
-      Rotation2d driveRotation = drive.getPose().getRotation();
-      if (DriverStation.getAlliance().isEmpty()
-          || DriverStation.getAlliance().get() == Alliance.Red) {
-        driveRotation = driveRotation.plus(new Rotation2d(Math.PI));
+    ChassisSpeeds speeds;
+    if (!robotRelativeOverride.get()) {
+      speeds =
+          new ChassisSpeeds(
+              linearVelocity.getX() * DriveConstants.MAX_LINEAR_SPEED,
+              linearVelocity.getY() * DriveConstants.MAX_LINEAR_SPEED,
+              rightX * MAX_ANGULAR_SPEED);
+    } else {
+      if (DriverStation.getAlliance().isPresent()
```

Comment:
AllianceFlipUtil.shouldFlip is `empty || red`.  This is `present && red`. Can you make AllianceFlipUtil.shouldFlip public and use it directly instead of duplicating the logic?